src/khoj/interface/web/chat.html

```diff
@@ -801,17 +801,20 @@
             if (overlayText == null) {
                 dropzone.classList.add('dragover');
                 var overlayText = document.createElement("div");
-                overlayText.innerHTML = "Select a file or drag + drop it here to share it with Khoj";
+                overlayText.innerHTML = "Select file(s) or drag + drop it here to share it with Khoj";
                 overlayText.className = "dropzone-overlay";
                 overlayText.id = "dropzone-overlay";
                 dropzone.appendChild(overlayText);
             }
 
             const fileInput = document.createElement('input');
             fileInput.type = 'file';
+            fileInput.multiple = true;
             fileInput.addEventListener('change', function() {
-                const selectedFile = fileInput.files[0];
-                uploadDataForIndexing(selectedFile);
+                const selectedFiles = fileInput.files;
+                for (let i = 0; i < selectedFiles.length; i++) {
+                    uploadDataForIndexing(selectedFiles[i]);
```

Comment:
Rather than sending each of the files individually for indexing, can you create a batch request? You should be able to do it by modifying `formData` in the `uploadDataForIndexing` method with data for each of the files. Process each file, and add it to `formData` like this, before passing `formData` to the indexing API. 

```
for (let file of files) {
    reader.onload = function (event) {
        fileContents = event.target.result;
        let fileType = file.type;
        if (fileType === "") {
            if (fileName.split('.').pop() === "org") {
                fileType = "text/org";
            } else if (fileName.split('.').pop() === "md") {
                fileType = "text/markdown";
            } else if (fileName.split('.').pop() === "txt") {
                fileType = "text/plain";
            } else if (fileName.split('.').pop() === "html") {
                fileType = "text/html";
            } else if (fileName.split('.').pop() === "pdf") {
                fileType = "application/pdf";
            }
        }

        let fileObj = new Blob([fileContents], { type: fileType });
        formData.append("files", fileObj, file.name);
    };
    reader.readAsText(file);
}
```

Reply:
Will fix this. Thanks!